marketplace/netlify-build/src/NetlifySideBarBuildButton.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'whatwg-fetch';
+import { Button, ValidationMessage } from '@contentful/forma-36-react-components';
+import { normalizeMessage, isOutOfOrder, isDuplicate, messageToState } from './message-processor';
+import { createPubSub } from './pubnub-client';
+
+import '@contentful/forma-36-react-components/dist/styles.css';
+import styles from './styles';
+
+import { EVENT_TRIGGERED, EVENT_TRIGGER_FAILED } from './contstants';
+
+export default class NeflifySideBarBuildButton extends React.Component {
+  static propTypes = {
+    site: PropTypes.object.isRequired,
+    users: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
+    userId: PropTypes.string.isRequired
+  };
+
+  state = { history: [] };
+
+  async componentDidMount() {
+    this.createPubSub();
+  }
+
+  componentDidUpdate(prevProps) {
+    // on initial render we may not get a user array
+    const hasNewUsers = !prevProps.users.length && this.props.users.length;
+
+    if (hasNewUsers || (this.props.site !== prevProps.site && this.pubsub)) {
```

Comment:
Do we need to stop and start if we get new users?

Reply:
Short answer: No.

But that would mean refactoring how the messaging system works. If you look on line 46, you'll see that there is a preconfigured message normalizer function that will get passed into the main listener which subscribes to pubnub messages. I could refactor that to be able to dynamically add in user info if some new info comes along but I figured it's just easier to stop it and start it up again. It's the same logic path we take if the `site` prop were to change and it seems to be reasonable there.